lib/fauna/client.rb

```diff
@@ -47,6 +47,7 @@ def initialize(params = {})
       @connection_timeout = params[:connection_timeout] || 60
       @observer = params[:observer]
       @adapter = params[:adapter] || :net_http_persistent
+      @last_seen = nil
```

Comment:
Should this be per-connection, as I have it, or somehow global? What's the concurrency story in this driver?

Reply:
I think it should be consistent across calls to `with_secret`.

Reply:
Hmm. Is #dup going to do that for me, or do I need to define #initialize_copy...

Reply:
I think you'll need to extract the `last_seen` to a class make sure dub only duplicates the reference.

Reply:
That's what dup does - it copies only the reference. Like this:

```ruby
class People
  attr_accessor :parent
end

me = People.new
dad = People.new

me.parent = dad
me.dup.parent.object_id == me.parent.object_id
```

Reply:
Since I need to put it under a mutex, a separate object makes sense. I'll knock that out.